### PR TITLE
feat: add watch for announcements and local testing

### PR DESCRIPTION
**Pull request by marcomariscal**
watch for announcements and assess if it's for the user.

also updates the tests to use a local fork when appropriate.

### CODE REVIEW

File: src/lib/helpers/test/setupTestEnv.ts [hunk 1]
```diff
@@ -39,13 +42,38 @@ function setupTestEnv() {
...
-export { getValidChainId };
+/**
+ * Retrieves the RPC URL based on the `useLocal` flag.
+ * @param {boolean} useLocal - Determines which environment variable to use for the RPC URL.
+ * @returns {string} The RPC URL.
+ * @throws {Error} If the corresponding environment variable is not defined.
+ */
+function getRpcUrl(useLocal = false) {
+  const rpcUrl = useLocal
+    ? process.env.TEST_LOCAL_NODE_RPC_URL
```
Comment (garyghayrat) [inline src/lib/helpers/test/setupTestEnv.ts:66]: Let's add this to `.env.example`

File: src/lib/helpers/test/setupTestEnv.ts [hunk 2]
```diff
@@ -2,17 +2,20 @@ import { createStealthClient } from '../..';
 import { ERC5564_CONTRACT, ERC6538_CONTRACT } from '../../..';
```
Comment (garyghayrat) [inline src/lib/helpers/test/setupTestEnv.ts:2]: I'm getting the following when `TEST_LOCAL_NODE_RPC_URL` is not defined.
image

But when I define it as the anvil local node, I get this
image

But when I change the chainId in `.env` to 31337, I get:
image

File: src/lib/helpers/test/setupTestEnv.ts (hunk 1 above)
Comment (marcomariscal) [reply src/lib/helpers/test/setupTestEnv.ts:66]: https://github.com/ScopeLift/stealth-address-sdk/pull/26/commits/efe0eb52edc106c60a0667646a9029f29e4b74e3

File: src/lib/helpers/test/setupTestEnv.ts (hunk 2 above)
Comment (marcomariscal) [reply src/lib/helpers/test/setupTestEnv.ts:2]: https://github.com/ScopeLift/stealth-address-sdk/pull/26/commits/e86150d49212370ccb31ddf6b25c575bd3c61c60

i updated the testing scripts and the readme to hopefully read more clearly on how to set up; still needs a bit more work to make it intuitive.

File: makefile [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+anvil-remote:
+	@echo "Starting Anvil with remote chain..."
+	# Start Anvil with the provided rpc url chain and capture its PID
+	anvil -f $(TEST_RPC_URL) & echo $$! > anvil_pid.txt
+
+# Make anvil local
+anvil-local:
+	@echo "Starting Anvil with local fork..."
+	# Start Anvil with the provided rpc url chain and capture its PID
+	anvil -f $(TEST_RPC_URL) & echo $$! > anvil_pid.txt
```
Comment (garyghayrat) [inline makefile:16]: This shouldn't fork the RPC right? It's currently the same as `anvil-remote`. Also I don't know about sending anvil into the background. It's convenient, but if we decide to keep it, then we should have a command that kills the anvil instance for the user by reading the file.

File: package.json [hunk 4]
```diff
@@ -6,9 +6,13 @@
   "module": "dist/index.js",
   "type": "module",
   "scripts": {
+    "anvil-local": "make anvil-local",
+    "anvil-remote": "make anvil-remote",
+    "test": "make test-local",
+    "test-remote": "make test-remote",
+    "test-local": "make test-local",
```
Comment (garyghayrat) [inline package.json:13]: currently the `test-local` doesn't actualy work with a local anvil instance.

File: package.json [hunk 5]
```diff
@@ -6,9 +6,13 @@
   "module": "dist/index.js",
   "type": "module",
   "scripts": {
+    "anvil-local": "make anvil-local",
```
Comment (garyghayrat) [inline package.json:9]: Can't we put the make file commands directly here and get rid of the makefile.

**Comment by marcomariscal** [comment]
> watch for announcements and assess if it's for the user.
> 
> also updates the tests to use a local fork when appropriate.

[tests are now defaulted to local.](https://github.com/ScopeLift/stealth-address-sdk/pull/37)

File: makefile (hunk 3 above)
Comment (garyghayrat) [reply makefile]: No worries! created an issue for it https://github.com/ScopeLift/stealth-address-sdk/issues/38

File: package.json (hunk 4 above)
Comment (garyghayrat) [reply package.json:12]: https://github.com/ScopeLift/stealth-address-sdk/pull/37

**Pull request merged**